presto-hive/src/main/java/io/prestosql/plugin/hive/AcidInfo.java

```diff
@@ -196,13 +276,39 @@ public Builder addDeleteDelta(Path deleteDeltaPath, long minWriteId, long maxWri
             return this;
         }
 
+        public Builder addOriginalFile(Path originalFilePath, long originalFileLength, int bucketId)
+        {
+            requireNonNull(originalFilePath, "originalFilePath is null");
+/*            Path partitionPathFromOriginalPath = originalFilePath.getParent();
+            checkArgument(
+                    partitionLocation.equals(partitionPathFromOriginalPath),
+                    "Partition location in OriginalFile '%s' does not match stored location '%s'",
+                    originalFilePath.getParent().toString(),
+                    partitionLocation);*/
+            bucketIdToOriginalFileInfoMap.put(bucketId, new OriginalFileInfo(originalFilePath.getName(), originalFileLength));
+            return this;
+        }
+
+        public Optional<AcidInfo> buildWithRequiredOriginalFiles(int bucketId)
+        {
+            // 1. Fetch list of all the original files which have same bucket Id
+            // 2. Build AcidInfo
+            checkState(bucketId > -1 && bucketIdToOriginalFileInfoMap.containsKey(bucketId), "Bucket Id to OriginalFileInfo map should have " +
+                    "entry for requested bucket Id");
+            List<DeleteDeltaInfo> deleteDeltas = deleteDeltaInfoBuilder.build();
+            if (deleteDeltas.isEmpty()) {
+                return Optional.empty();
```

Comment:
With current shape of this PR Presto fails to read table data if there are original files and there are **no** delete_delta files present.
If `(deleteDeltas.isEmpty())` resolves to `true` we are loosing all the original files information.  Yet Orc reader, later on, expects the ACID columns to be present in data files.
This results in following exception on read:
```
Query 20200720_104400_00016_v7svz failed: ORC ACID file should have 6 columns: hdfs://hadoop-master:9000/user/hive/warehouse/t/000000_0
io.prestosql.spi.PrestoException: ORC ACID file should have 6 columns: hdfs://hadoop-master:9000/user/hive/warehouse/t/000000_0
        at io.prestosql.plugin.hive.orc.OrcPageSourceFactory.verifyAcidSchema(OrcPageSourceFactory.java:418)
        at io.prestosql.plugin.hive.orc.OrcPageSourceFactory.createOrcPageSource(OrcPageSourceFactory.java:244)
        at io.prestosql.plugin.hive.orc.OrcPageSourceFactory.createPageSource(OrcPageSourceFactory.java:160)
        at io.prestosql.plugin.hive.HivePageSourceProvider.createHivePageSource(HivePageSourceProvider.java:178)
        at io.prestosql.plugin.hive.HivePageSourceProvider.createPageSource(HivePageSourceProvider.java:105)
        at io.prestosql.plugin.base.classloader.ClassLoaderSafeConnectorPageSourceProvider.createPageSource(ClassLoaderSafeConnectorPageSourceProvider.java:57)
        at io.prestosql.split.PageSourceManager.createPageSource(PageSourceManager.java:64)
        at io.prestosql.operator.TableScanOperator.getOutput(TableScanOperator.java:298)
        at io.prestosql.operator.Driver.processInternal(Driver.java:379)
        at io.prestosql.operator.Driver.lambda$processFor$8(Driver.java:283)
        at io.prestosql.operator.Driver.tryWithLock(Driver.java:675)
        at io.prestosql.operator.Driver.processFor(Driver.java:276)
        at io.prestosql.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:1075)
        at io.prestosql.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:163)
        at io.prestosql.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:484)
        at io.prestosql.$gen.Presto_unknown____20200720_103058_2.run(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
```

Repro steps:
Create non-transactional table on hive and add some data:
```
  CREATE TABLE t (a integer) stored as orc tblproperties ('transactional'='false');
  insert into t values (1);
```

make table transactional:
```
ALTER TABLE t SET TBLPROPERTIES ('transactional'='true');
```

At this point reads from presto fail. To make them work we currently need to delete something from table t, so delete_delta file is present.

Reply:
@losipiuk : Thank you for pointing this out, will fix this.

Reply:
```diff
diff --git a/presto-hive/src/main/java/io/prestosql/plugin/hive/AcidInfo.java b/presto-hive/src/main/java/io/prestosql/plugin/hive/AcidInfo.java
index 15838b165b..14f8d8d580 100644
--- a/presto-hive/src/main/java/io/prestosql/plugin/hive/AcidInfo.java
+++ b/presto-hive/src/main/java/io/prestosql/plugin/hive/AcidInfo.java
@@ -48,7 +48,6 @@ public class AcidInfo
     {
         this.partitionLocation = requireNonNull(partitionLocation, "partitionLocation is null");
         this.deleteDeltas = ImmutableList.copyOf(requireNonNull(deleteDeltas, "deleteDeltas is null"));
-        checkArgument(!deleteDeltas.isEmpty(), "deleteDeltas is empty");
         this.originalFiles = ImmutableList.copyOf(requireNonNull(originalFiles, "originalFiles is null"));
         this.bucketId = bucketId;
     }
@@ -296,9 +295,6 @@ public class AcidInfo
             checkState(bucketId > -1 && bucketIdToOriginalFileInfoMap.containsKey(bucketId), "Bucket Id to OriginalFileInfo map should have " +
                     "entry for requested bucket Id");
             List<DeleteDeltaInfo> deleteDeltas = deleteDeltaInfoBuilder.build();
-            if (deleteDeltas.isEmpty()) {
-                return Optional.empty();
-            }
             return Optional.of(new AcidInfo(partitionLocation.toString(), deleteDeltas, bucketIdToOriginalFileInfoMap.get(bucketId), bucketId));
         }
```

seems to improve situation, yet I am not sure if it does not break some flow.